common/firmware/system/app_update.c

```diff
@@ -23,9 +24,24 @@ void app_update_start() {
     // Notify bootloader that an update is requested by main application.
     firmware_update_flags = UPDATE_FLAG_REQUESTED;
 
+    // Clear reset flags for bootloader.
+    __HAL_RCC_CLEAR_RESET_FLAGS();
```

Comment:
This line resets the `RCC_CSR` register. It hides the reason for a reset. 

Why here? Why now? 

We're about to jump to the bootloader. The bootloader will not check `firmware_update_flags` when on a system reset (`firmware_update_flags` is an uniitialized memory location).

Reply:
Oh, I disagree with this at least for future work. The bootloader detects (or will detect) the reset type to determine if it's running a) because it's the first thing that runs after power up and everything is fine or b) if the app crashed or something. If we only reset this value here, then if we crash before this what will the bootloader see?

Reply:
Are you thinking that the register should be cleared by the bootloader or not at all?